File: src/main/java/com/itszuvalex/femtocraft/power/tiles/TileEntityCryoEndothermalChargingBase.java
package com.itszuvalex.femtocraft.power.tiles;

import com.itszuvalex.femtocraft.api.EnumTechLevel;
import com.itszuvalex.femtocraft.api.power.ICryoEndothermalChargingBase;
import com.itszuvalex.femtocraft.api.power.PowerContainer;
import net.minecraftforge.common.util.ForgeDirection;

/**
 * Created by Chris on 9/8/2014.
 */
public class TileEntityCryoEndothermalChargingBase extends TileEntityPowerProducer implements
        ICryoEndothermalChargingBase {
    public static final int maxStorage = 25000;
    public static final EnumTechLevel powerLevel = EnumTechLevel.NANO;

    @Override
    public PowerContainer defaultContainer() {
        return new PowerContainer(powerLevel, maxStorage);
    }

    @Override
    public int addPower(int power) {
        return charge(ForgeDirection.DOWN, power);
    }

    @Override
    public boolean isTechLevelSupported(EnumTechLevel tech) {
        return powerLevel == tech;
    }
}

File: src/main/java/com/itszuvalex/femtocraft/power/tiles/TileEntityCryoEndothermalChargingBase.scala
package com.itszuvalex.femtocraft.power.tiles

import com.itszuvalex.femtocraft.api.EnumTechLevel
import com.itszuvalex.femtocraft.api.core.Configurable
import com.itszuvalex.femtocraft.api.power.{ICryoEndothermalChargingBase, PowerContainer}
import com.itszuvalex.femtocraft.core.tiles.TileEntityBase
import com.itszuvalex.femtocraft.power.tiles.TileEntityCryoEndothermalChargingBase._
import com.itszuvalex.femtocraft.power.traits.PowerProducer
import net.minecraftforge.common.util.ForgeDirection

/**
 * Created by Chris on 9/8/2014.
 */
object TileEntityCryoEndothermalChargingBase {
  @Configurable(comment = "Power capacity.")     val maxStorage = 25000
  @Configurable(comment = "Tech level of power") val powerLevel = EnumTechLevel.NANO
  @Configurable(comment = "Maximum number of addons supported") val maximumDepth = 15
}

@Configurable class TileEntityCryoEndothermalChargingBase extends TileEntityBase with PowerProducer with ICryoEndothermalChargingBase {
  override def defaultContainer = new PowerContainer(powerLevel, maxStorage)

  override def addPower(power: Int) = charge(ForgeDirection.DOWN, power)

  override def isTechLevelSupported(tech: EnumTechLevel) = powerLevel eq tech

  override def maximumDepthSupported() = maximumDepth
}
